make/RunTests.gmk

```diff
@@ -715,38 +715,37 @@ endef
 #
 SetupAot = $(NamedParamsMacroTemplate)
 define SetupAotBody
+  $1_AOT_JDK_DIR   := $$($1_TEST_SUPPORT_DIR)/aot
   $1_AOT_JDK_CONF  := $$($1_TEST_SUPPORT_DIR)/aot/jdk.aotconf
   $1_AOT_JDK_CACHE := $$($1_TEST_SUPPORT_DIR)/aot/jdk.aotcache
 
-  $1_JAVA_TOOL_OPTS := $$(addprefix -J, $$($1_VM_OPTIONS))
+  # We first execute the training run with make/test/SetupAot.java to touch
+  # a fair number of classes inside the JDK. Note that we don't specify a classpath,
+  # but the Java launcher will use an implicit class path of ".", so it can pick up the
+  # SetupAot class from the current directory.
+  #
+  # The SetupAot class (or any other classes that are loaded from "."
+  # will be excluded from the the AOT cache as "." is an unsupported
+  # location. As a result, the AOT cache will contain only classes from the JDK.
 
   $$($1_AOT_JDK_CACHE): $$(JDK_IMAGE_DIR)/release
 	$$(call MakeDir, $$($1_TEST_SUPPORT_DIR)/aot)
 
-	$(foreach jtool, javac javap jlink jar, \
-	    $(info AOT: Create cache configuration for $(jtool)) \
-	    $$(call ExecuteWithLog, $$($1_TEST_SUPPORT_DIR)/aot.$(jtool), ( \
-	        $$(FIXPATH) $(JDK_UNDER_TEST)/bin/$(jtool) $$($1_JAVA_TOOL_OPTS) \
-	                    -J-XX:AOTMode=record -J-XX:AOTConfiguration=$$($1_AOT_JDK_CONF).$(jtool) --help \
-	    ))
-	)
-
-	$$(info AOT: Copy $(JDK_UNDER_TEST)/lib/classlist to $$($1_AOT_JDK_CONF).jdk )
+	$(info AOT: Create cache configuration) \
 	$$(call ExecuteWithLog, $$($1_TEST_SUPPORT_DIR)/aot, ( \
-		$$(FIXPATH) $(CP) $(JDK_UNDER_TEST)/lib/classlist $$($1_AOT_JDK_CONF).jdk \
+	    $(CD) $$($1_AOT_JDK_DIR); \
+	    $(JDK_UNDER_TEST)/bin/javac -d . $(TOPDIR)/make/test/SetupAot.java; \
+	    $$(FIXPATH) $(JDK_UNDER_TEST)/bin/java $$($1_VM_OPTIONS) \
+	                -Xlog:cds,cds+class=debug:file=$$($1_AOT_JDK_CONF).log \
+	                -XX:AOTMode=record -XX:AOTConfiguration=$$($1_AOT_JDK_CONF) \
+	                SetupAot \
```

Comment:
Please use 4 spaces indent to follow the [Code Conventions for the Build System](https://openjdk.org/groups/build/doc/code-conventions.html).

```suggestion
	    $$(FIXPATH) $(JDK_UNDER_TEST)/bin/java $$($1_VM_OPTIONS) \
	        -Xlog:cds,cds+class=debug:file=$$($1_AOT_JDK_CONF).log \
	        -XX:AOTMode=record -XX:AOTConfiguration=$$($1_AOT_JDK_CONF) \
	        SetupAot \
```

Reply:
Fixed.